purchases/src/test/java/com/revenuecat/purchases/BasePurchasesTest.kt

```diff
@@ -0,0 +1,355 @@
+            } just Runs
+            every {
+                mockCache.cacheOfferings(any())
+            } just Runs
+        }
+    }
+
+    private fun mockSubscriberAttributesManager() {
+        every {
+            mockSubscriberAttributesManager.synchronizeSubscriberAttributesForAllUsers(appUserId)
+        } just Runs
+    }
+    // endregion
+
+    // region Protected methods
+    protected fun mockCustomerInfoHelper(errorGettingCustomerInfo: PurchasesError? = null) {
+        with(mockCustomerInfoHelper) {
+            every {
+                retrieveCustomerInfo(any(), any(), false, any())
+            } answers {
+                val callback = arg<ReceiveCustomerInfoCallback?>(3)
+                if (errorGettingCustomerInfo == null) {
+                    callback?.onReceived(mockInfo)
+                } else {
+                    callback?.onError(errorGettingCustomerInfo)
+                }
+            }
+            every {
+                cacheCustomerInfo(any())
+            } just runs
+            every {
+                sendUpdatedCustomerInfoToDelegateIfChanged(any())
+            } just runs
+            every {
+                updatedCustomerInfoListener = any()
+            } just runs
+            every {
+                updatedCustomerInfoListener
+            } returns null
+        }
+    }
+
+    protected fun mockProducts(response: String = ONE_OFFERINGS_RESPONSE) {
+        every {
+            mockBackend.getOfferings(any(), any(), captureLambda(), any())
+        } answers {
+            lambda<(JSONObject) -> Unit>().captured.invoke(JSONObject(response))
+        }
+    }
+
+    protected fun mockStoreProduct(
+        productIds: List<String>,
+        productIdsSuccessfullyFetched: List<String>,
+        type: ProductType
+    ): List<StoreProduct> {
+        val storeProducts = productIdsSuccessfullyFetched.map { productId ->
+            if (type == ProductType.SUBS) stubStoreProduct(productId, stubSubscriptionOption("p1m", "P1M"))
+            else createMockOneTimeProductDetails(productId).toInAppStoreProduct()
+        }.mapNotNull { it }
+
+        every {
+            mockBillingAbstract.queryProductDetailsAsync(
+                type,
+                productIds.toSet(),
+                captureLambda(),
+                any()
+            )
+        } answers {
+            lambda<(List<StoreProduct>) -> Unit>().captured.invoke(storeProducts)
+        }
+        return storeProducts
+    }
+
+    protected fun getMockedStoreTransaction(
+        productId: String,
+        purchaseToken: String,
+        productType: ProductType
+    ): StoreTransaction {
+        val p: PurchaseHistoryRecord = stubPurchaseHistoryRecord(
+            productIds = listOf(productId),
+            purchaseToken = purchaseToken
+        )
+
+        return p.toStoreTransaction(productType)
+    }
+
+    protected fun buildPurchases(anonymous: Boolean, autoSync: Boolean = true) {
+        purchases = Purchases(
```

Comment:
I think the only other place we build this in tests currently is in `SubscriberAttributesPurchasesTests`. Might be good trying to have that use this `BasePurchasesTest`. But we can do that as a followup.

Reply:
let's do it as a followup, otherwise the pr gets harder to review